src/common/bip32.c

```diff
@@ -20,12 +20,20 @@ bool bip32_path_read(const uint8_t *in, size_t in_len, bip32_path_t *out) {
             return false;
         }
         out->path[i] = read_u32_be(in, offset);
-        if ((out->path[i] & 0x7FFFFFFFu) > MAX_DERIVATION_INDEX) {
+        offset += 4;
+    }
+    if (out->length > 4) {
+        if ((out->path[4] & 0x7FFFFFFFu) > MAX_DERIVATION_INDEX) {
```

Comment:
Why are we checking only `path[4]`? What about the others? `path[5], path[6], ...`?

Reply:
`path[4]` is the address_index level (the fifth derivation index) and since we allow only 5 derivation indexes there would be no `path[5]` or 6.

Reply:
Ok. Let's assert that `out->length < 6`. We can either raise an exception or return false.

Reply:
You just showed me that we are already doing it. Thanks!